kernel/src/scan/data_skipping.rs

```diff
@@ -41,6 +41,7 @@ fn as_data_skipping_predicate(expr: &Expr, inverted: bool) -> Option<Expr> {
     DataSkippingPredicateCreator.eval_expr(expr, inverted)
 }
 
+#[derive(Clone)]
```

Comment:
why clone?

Reply:
I need a data skipping filter for each commit, so I clone the filter. 

This happens because the replay scanner only works on a commit granularity for CDf, whereas the snapshot replay scanner can use a single filter for the entire log.

Given that everything is Arc, this should be relatively cheap.

Reply:
Now that we construct `DataSkippingVisitor` for each commit, we no longer need the clone 👍